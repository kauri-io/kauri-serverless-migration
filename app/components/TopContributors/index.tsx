import { makeStyles, Theme, Typography } from '@material-ui/core'
import Avatar from '../Avatar'

const useStyles = makeStyles((theme: Theme) => ({
    title: {
        fontWeight: 400,
        padding: theme.spacing(2, 0),
        lineHeight: '24px',
    },
    container: {
        background: theme.palette.common.white,
        padding: theme.spacing(2),
        borderRadius: 4,
        width: '100%',
        display: 'flex',
        flexDirection: 'column',
        '& > *': {
            marginBottom: theme.spacing(2),
        },
    },
    subtitle: {},
    root: {
        [theme.breakpoints.down('md')]: {
            display: 'none',
        },
    },
}))

export default ({ contributors }) => {
    const classes = useStyles({})
    return (
        <div className={classes.root}>
            <Typography variant="h6" component="h3" className={classes.title}>
                Top Contributors
            </Typography>
            <div className={classes.container}>
                <Typography className={classes.subtitle} variant="body2">
                    Contributors of the week
                </Typography>
                {contributors.map((contributor, index) => (
                    <Avatar
                        key={index}
                        withName={true}
                        username={contributor.username}
                        id={contributor.userId}
                        avatar={contributor.avatar}
                    />
                ))}
            </div>
        </div>
    )
}
