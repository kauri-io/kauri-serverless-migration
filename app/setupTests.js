import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
