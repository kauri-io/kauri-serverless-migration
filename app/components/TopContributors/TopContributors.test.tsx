import TopContributors from './index'
import { mountWithTheme } from '../../setupTests'

let props = {
    contributors: [
        {
            avatar: 'test url',
            username: 'test username',
        },
    ],
    routeChangeAction: jest.fn(),
    linkComponent: jest.fn(),
}

describe('components/TopContributors', () => {
    it('snapshot should match', () => {
        const wrapper = mountWithTheme(<TopContributors {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})
