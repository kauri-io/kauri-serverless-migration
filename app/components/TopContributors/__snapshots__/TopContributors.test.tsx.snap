// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TopContributors snapshot should match 1`] = `
<ThemeProvider
  theme={
    Object {
      "DEFAULT_CARD_WIDTH": 288,
      "MAIN_BODY_WIDTH": 1242,
      "bg": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "bgPrimary": "#1E2428",
      "bgSecondary": "#1E3D3B",
      "borderTextArea": "#979797",
      "breakpoints": Array [
        "500px",
        "52em",
        "1242px",
      ],
      "colors": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
      "disabledBackgroundColor": "#D6D6D6",
      "disabledTextColor": "#9B9B9B",
      "divider": "#EBEBEB",
      "errorRedColor": "#C03030",
      "fontSizes": Array [
        11,
        13,
        14,
        16,
        18,
        20,
        22,
        28,
        32,
      ],
      "fontWeight": Array [
        300,
        "normal",
        500,
        700,
      ],
      "hoverTextColor": "#108E72",
      "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
      "padding": "calc((100vw - 1242px) / 2)",
      "paddingTop": "2.5em",
      "primary": "#0BA986",
      "primaryColor": "#0BA986",
      "primaryDark": "#11856B",
      "primaryTextColor": "#1E2428",
      "secondary": "#11856B",
      "secondaryBlue": "rgb(81, 123, 162, 1)",
      "secondaryBlueDark": "rgb(81, 118, 142, 1)",
      "secondaryColor": "#1E3D3B",
      "secondaryTextColor": "#283035",
      "space": Array [
        0,
        10,
        15,
        30,
        60,
      ],
      "tertiaryBackgroundColor": "#F7F7F7",
      "textPrimary": "#1E2428",
      "white": "#FFFFFF",
    }
  }
>
  <TopContributors
    contributors={
      Array [
        Object {
          "avatar": "test url",
          "username": "test username",
        },
      ]
    }
    linkComponent={
      [MockFunction] {
        "calls": Array [
          Array [
            <UserAvatarComponent
              avatar="test url"
              username="test username"
            />,
            "/public-profile/undefined",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    routeChangeAction={[MockFunction]}
  >
    <TopResourcesSection>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "TopResourcesSection-mr7g3s-0",
              "isStatic": false,
              "lastClassName": "bYCQSD",
              "rules": Array [
                "display:flex;flex-direction:column;>:first-child{margin-bottom:",
                [Function],
                "px;}>:nth-child(2){margin-top:0px;}> * > *:nth-child(n + 6){display:none;}@media (max-width:",
                [Function],
                "){display:none;}",
              ],
            },
            "displayName": "TopResourcesSection",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "TopResourcesSection-mr7g3s-0",
            "target": "section",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <section
          className="TopResourcesSection-mr7g3s-0 bYCQSD"
        >
          <Typography__styledComponent>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "Typography__styledComponent-sc-36xkf6-5",
                    "isStatic": false,
                    "lastClassName": "eLGQne",
                    "rules": Array [
                      "color:",
                      [Function],
                      ";margin:0px;font-weight:",
                      "500",
                      ";",
                      ";font-size:",
                      "20",
                      "px;text-transform:",
                      [Function],
                      ";text-align:",
                      [Function],
                      ";:hover{color:",
                      [Function],
                      ";cursor:",
                      [Function],
                      ";}",
                    ],
                  },
                  "displayName": "Typography__styledComponent",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "Typography__styledComponent-sc-36xkf6-5",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <span
                className="Typography__styledComponent-sc-36xkf6-5 eLGQne"
              >
                Top Contributors
              </span>
            </StyledComponent>
          </Typography__styledComponent>
          <TopContributors__ContributorsContainer>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "TopContributors__ContributorsContainer-sc-2nkhip-0",
                    "isStatic": false,
                    "lastClassName": "cOmYpI",
                    "rules": Array [
                      "display:flex;flex-direction:column;>:not(:last-child){margin-bottom:",
                      [Function],
                      "px;}",
                    ],
                  },
                  "displayName": "TopContributors__ContributorsContainer",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "TopContributors__ContributorsContainer-sc-2nkhip-0",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="TopContributors__ContributorsContainer-sc-2nkhip-0 cOmYpI"
              />
            </StyledComponent>
          </TopContributors__ContributorsContainer>
        </section>
      </StyledComponent>
    </TopResourcesSection>
  </TopContributors>
</ThemeProvider>
`;
